vumi/persist/fields.py

```diff
@@ -957,3 +963,27 @@ class ManyToMany(ForeignKey):
 
     def __init__(self, other_model, index=None, backlink=None):
         super(ManyToMany, self).__init__(other_model, index, backlink)
+
+
+class ComputedIndexDescriptor(FieldDescriptor):
+    """A field descriptor for computed index fields."""
+
+    def pre_save(self, modelobj):
+        self.set_value(modelobj, self.field.value_func(modelobj))
+
+    def _add_index(self, modelobj, value):
+        # We override `FieldDescriptor._add_index` because we don't want `None`
+        # values.
+        if value is not None:
+            modelobj._riak_object.add_index(self.index_name, str(value))
```

Comment:
`set_value()` clears the index values and then calls `_add_index()` as necessary.